src/lib/comms/sol-coap.c

```diff
@@ -1438,6 +1457,19 @@ sol_coap_server_new(int port)
 }
 
 SOL_API int
+sol_coap_packet_get_buf(struct sol_coap_packet *pkt, uint8_t **buf, uint16_t *len)
```

Comment:
I've discussed with vcgomes and he'll modify it in such a way accessing buf won't be required.
So I'll wait for this solution and rebase to avoid exposing this function.
Thanks